### PR TITLE
Add support for profiling on Python 3.8.

### LINKED ISSUE #5341: Null argument error when selecting other interpeter in profiling dialog
Start VS
Launch Python profiling dialog
Select "Other" interpreter
Put path to python.exe from Python 2.7
Put full path to script pystone.py
Press OK

image

Result: null path error

This is a regression in the code that checks to activate conda environment.

### LINKED ISSUE #5683: Support profiling on Python 3.8
```
Traceback (most recent call last):
  File "c:\users\huvalo\appdata\local\microsoft\visualstudio\16.0_b2fdd7deexp\extensions\microsoft corporation\python - profiling\16.0.0\proflaun.py", line 35, in <module>
    vspyprof.profile(__file__, globals(), locals(), profdll)
  File "c:\users\huvalo\appdata\local\microsoft\visualstudio\16.0_b2fdd7deexp\extensions\microsoft corporation\python - profiling\16.0.0\vspyprof.py", line 74, in profile
    raise NotImplementedError("Profiling is currently not supported for " + sys.version)
NotImplementedError: Profiling is currently not supported for 3.8.0b3 (tags/v3.8.0b3:4336222, Jul 29 2019, 16:33:20) [MSC v.1916 32 bit (Intel)]
Press any key to continue . . .
```

### PR DESCRIPTION
**Pull request by huguesv**
Fix #5683 
Fix #5341

I added version specific project profiling tests, to ensure that Python 2.7...3.8 are covered (and the 3.8 test fails without my changes)

image

### CODE REVIEW

File: Python/Product/VsPyProf/python.h [hunk 1]
```diff
@@ -231,15 +233,47 @@ class PyCodeObject37 : public PyObject {
...
+// 3.8
+class PyCodeObject38 : public PyObject {
+public:
+    int co_argcount;            /* #arguments, except *args */
+    int co_posonlyargcount;     /* #positional only arguments */
+    int co_kwonlyargcount;      /* #keyword only arguments */
+    int co_nlocals;             /* #local variables */
+    int co_stacksize;           /* #entries needed for evaluation stack */
+    int co_flags;               /* CO_..., see below */
+    int co_firstlineno;         /* first source line number */
```
Comment (bschnurr) [inline Python/Product/VsPyProf/python.h]: shouldn't we initialize all these pointers below to null?

File: Python/Product/VsPyProf/python.h [hunk 2]
```diff
@@ -231,15 +233,47 @@ class PyCodeObject37 : public PyObject {
...
-        return version >= PythonVersion_37;
+        return version == PythonVersion_37;
+    }
+};
+
+// 3.8
+class PyCodeObject38 : public PyObject {
+public:
+    int co_argcount;            /* #arguments, except *args */
+    int co_posonlyargcount;     /* #positional only arguments */
```
Comment (huguesv) [inline Python/Product/VsPyProf/python.h]: @RaymonGulati1 fyi this `co_posonlyargcount` is new in 3.8

File: Python/Product/VsPyProf/python.h (hunk 1 above)
Comment (bschnurr) [reply Python/Product/VsPyProf/python.h]: in PythonApi.cpp I see null checking but these could be pointing to garbage

  argNames = (PyTupleObject*)((PyCodeObject25_27*)codeObj)->co_varnames;

if (argCount != 0 && argNames && argNames->ob_type == PyTuple_Type) {

Comment (huguesv) [reply Python/Product/VsPyProf/python.h]: no, we never instantiate this, we only need the structure of the class so we can offset correctly into existing objects in memory

Comment (bschnurr) [reply Python/Product/VsPyProf/python.h]: or is there some kind of special allocator for these?

**Pull request merged**